okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpClientTransport.java

```diff
@@ -356,10 +351,20 @@ public void run() {
       try {
         // Read until the underlying socket closes.
         while (frameReader.nextFrame(this)) {
```

Comment:
I see these cases:
 1. Server closes TCP connection. nextFrame will return false and we exit the loop
 2. Server sends another HTTP/2 frame (although I have no clue why the server would do that, since all the streams were closed) after we have closed frameWriter. nextFrame then completes and we noticed that stopped is now true so we exit the loop.
 3. The connection is idle. I don't see how this is cleaned up. Are there some internals of OkHttp that cause this case not to occur?